tests/meta/transfer/test_external_data.py

```diff
@@ -0,0 +1,210 @@
+):
+    """Test generating the source path"""
+
+    # WHEN the source path is created
+
+    source_path = external_data_api.create_source_path(
+        raw_path="/path/%s",
+        ticket_id=ticket_nr,
+        customer=customer_id,
+        cust_sample_id=cust_sample_id,
+    )
+
+    # THEN the source path is
+    assert source_path == Path("/path/cust000/123456/child/")
+
+
+def test_create_destination_path(
+    external_data_api: ExternalDataAPI, customer_id: str, sample_id: str
+):
+    """Test generating the destination path"""
+
+    # WHEN the source path is created
+    destination_path = external_data_api.create_destination_path(
+        raw_path="/path/%s", customer=customer_id, lims_sample_id=sample_id
+    )
+
+    # THEN the source path is
+    assert destination_path == Path("/path/cust000/ADM1/")
+
+
+def test_download_sample(
+    external_data_api: ExternalDataAPI,
+    mocker,
+    ticket_nr: int,
+    customer_id: str,
+    cust_sample_id: str,
+    sample_id: str,
+):
+    """Test for downloading external data via slurm"""
+
+    # GIVEN paths needed to run rsync
+    mocker.patch.object(ExternalDataAPI, "create_log_dir")
+    ExternalDataAPI.create_log_dir.return_value = Path("/path/to/log")
+
+    mocker.patch.object(ExternalDataAPI, "create_source_path")
+    ExternalDataAPI.create_source_path.return_value = Path("/path/to/source")
+
+    mocker.patch.object(ExternalDataAPI, "create_destination_path")
+    ExternalDataAPI.create_destination_path.return_value = Path("/path/to/destination")
+
+    # WHEN the destination path is created
+    sbatch_number = external_data_api.transfer_sample(
+        cust_sample_id=cust_sample_id,
+        ticket_id=ticket_nr,
+        cust=customer_id,
+        lims_sample_id=sample_id,
+        dry_run=True,
+    )
+
+    # THEN check that an integer was returned as sbatch number
+    assert isinstance(sbatch_number, int)
+
+
+def test_get_all_fastq(
+    cg_context: CGConfig, external_data_directory, external_data_api: ExternalDataAPI
+):
+    """Test the finding of fastq.gz files in customer directories."""
+    for folder in external_data_directory.iterdir():
+        # WHEN the list of file-paths is created
+        files = external_data_api.get_all_fastq(
+            sample_folder=external_data_directory.joinpath(folder)
+        )
+        # THEN only fast.gz files are returned
+        assert [tmp.suffixes == [".fastq", ".gz"] for tmp in files]
+
+
+def test_check_bundle_fastq_files(
+    external_data_api: ExternalDataAPI, sample_id: str, hk_version_obj
+):
+
+    # GIVEN a housekeeper bundle with a file
+    # mocker.patch.object(MockHousekeeperAPI, "get_files")
+    # MockHousekeeperAPI.get_files.return_value = [Path("/boring/old/boomer/file.fastq.gz")]
+    external_data_api.housekeeper_api.add_file(
+        path=Path("/boring/old/boomer/file.fastq.gz"), tags=["fastq"], version_obj=hk_version_obj
```

Comment:
Path("/boring/old/boomer/file.fastq.gz") is used twice - make it a fixture

Reuse the HK constant tag also in the test. That way if the tag changes this test will tell us and we now we have to modify the sub

Reply:
Due to restructuring this test has moved to tests/apps/hk/test_file.py and has been rewritten with more fixtures.